internal/impl/gcp/output_pubsub.go

```diff
@@ -52,221 +40,294 @@ Or delete all keys with:
-			docs.FieldObject("flow_control", "For a given topic, configures the PubSub client's internal buffer for messages to be published.").
-				WithChildren(
-					docs.FieldInt("max_outstanding_messages", "Maximum number of buffered messages to be published. If less than or equal to zero, this is disabled.").HasDefault(1000),
-					docs.FieldInt("max_outstanding_bytes", "Maximum size of buffered messages to be published. If less than or equal to zero, this is disabled.").HasDefault(-1),
-					docs.FieldString("limit_exceeded_behavior", "Configures the behavior when trying to publish additional messages while the flow controller is full. The available options are ignore (disable, default), block, and signal_error (publish results will return an error).").
-						HasDefault("ignore").
-						HasOptions("ignore", "block", "signal_error"),
-				).Advanced(),
-		).ChildDefaultAndTypesFromStruct(output.NewGCPPubSubConfig()),
-		Categories: []string{
-			"Services",
-			"GCP",
-		},
-	})
-	if err != nil {
-		panic(err)
-	}
+`+"```"+``).
+		Fields(
+			service.NewStringField("project").Description("The project ID of the topic to publish to."),
+			service.NewInterpolatedStringField("topic").Description("The topic to publish to."),
+			service.NewStringField("endpoint").
+				Default("").
+				Example("us-central1-pubsub.googleapis.com:443").
+				Example("us-west3-pubsub.googleapis.com:443").
+				Description("An optional endpoint to override the default of `pubsub.googleapis.com:443`. This can be used to connect to a region specific pubsub endpoint. For a list of valid values check out [this document.](https://cloud.google.com/pubsub/docs/reference/service_apis_overview#list_of_regional_endpoints)"),
+			service.NewInterpolatedStringField("ordering_key").
+				Optional().
+				Description("The ordering key to use for publishing messages.").
+				Advanced(),
+			service.NewIntField("max_in_flight").Default(0).Description("The maximum number of messages to have in flight at a given time. Increasing this may improve throughput."),
+			service.NewIntField("count_threshold").
+				Default(defaults.CountThreshold).
+				Description("Publish a pubsub buffer when it has this many messages"),
+			service.NewDurationField("delay_threshold").
+				Default(defaults.DelayThreshold.String()).
+				Description("Publish a non-empty pubsub buffer after this delay has passed."),
+			service.NewIntField("byte_threshold").
+				Default(defaults.ByteThreshold).
+				Description("Publish a batch when its size in bytes reaches this value."),
+			service.NewDurationField("publish_timeout").
+				Default(defaults.Timeout.String()).
+				Example("10s").
+				Example("5m").
+				Example("60m").
+				Description("The maximum length of time to wait before abandoning a publish attempt for a message.").
+				Advanced(),
+			service.NewMetadataFilterField("metadata").
+				Optional().
+				Description("Specify criteria for which metadata values are sent as attributes."),
```

Comment:
This is a breaking change because the old api used `metadata.exclude_prefixes` where is the new api uses `metadata.include_prefixes` and `metadata.include_patterns`. Should I manually implement the old config somehow?

Reply:
I might expand the public field definition so that you can toggle exclusive versus inclusive defaults. Ideally all components will eventually converge but we need to do it in stages (with backwards compatibility).